### PR TITLE
Support for SheetIndex Partition and Model

### PR DESCRIPTION
**Pull request by naveedkhan8067**
SheetIndex Partition and Model support is added for newer versions (01.00.16) of BisCore schema.

### CODE REVIEW

**Review by ColinKerr — APPROVED** [review]

File: iModelCore/iModelPlatform/DgnCore/DgnModel.cpp [hunk 1]
```diff
@@ -624,6 +624,48 @@ DgnDbStatus DefinitionModel::_OnInsertElement(DgnElementR el)
...
+    SheetIndexModelPtr model = Create(modeledElement);
+    return (model.IsValid() && (DgnDbStatus::Success == model->Insert())) ? model : nullptr;
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+DgnDbStatus SheetIndexModel::_OnInsertElement(DgnElementR element)
+    {
+    return element.IsInformationContentElement() ? T_Super::_OnInsertElement(element) : DgnDbStatus::WrongModel;
```
Comment (pmconne) [inline iModelCore/iModelPlatform/DgnCore/DgnModel.cpp:666]: This implementation is identical to that of the [superclass method](https://github.com/iTwin/imodel-native/blob/900ca418ffc8e2c1c34156bfa2356f9f1492d49e/iModelCore/iModelPlatform/DgnCore/DgnModel.cpp#L562).

Comment (naveedkhan8067) [reply iModelCore/iModelPlatform/DgnCore/DgnModel.cpp]: Function removed.

**Comment by kabentley** [comment]
I don't understand why this is necessary. Where do we ever use the handlers?

File: iModelCore/iModelPlatform/Tests/DgnProject/NonPublished/DgnModel_Test.cpp [hunk 2]
```diff
@@ -656,6 +657,41 @@ TEST_F(DgnModelTests, DefinitionModelCreation)
 /*---------------------------------------------------------------------------------**//**
 * @bsimethod
 +---------------+---------------+---------------+---------------+---------------+------*/
+TEST_F(DgnModelTests, SheetIndexModelCreation)
```
Comment (kabentley) [inline iModelCore/iModelPlatform/Tests/DgnProject/NonPublished/DgnModel_Test.cpp:660]: this test should be done in TypeScript and all of this "handler" stuff should be removed.

**Comment by naveedkhan8067** [comment]
> I don't understand why this is necessary. Where do we ever use the handlers? Isn't this whole PR totally useless?

The [PR ](https://github.com/iTwin/bis-schemas/pull/196) for the release of BisCore.1.0.16 is blocked by a failing test `ECInstancesCompatibility.InstancesCompatibilitySeed` from `iModelPlatform`. Its failing with `DgnDbStatus::WrongModel` status for following new classes: `SheetIndex`, `SheetIndexFolder`, `SheetIndexReference`, and `SheetReference` while inserting an element. This test has following 3 cases:

```
if (className == BIS_CLASS_Subject)
{
 ECN::ECClassCP relClass = ecClass->GetSchema().GetClassCP(BIS_REL_SubjectOwnsSubjects);
 ASSERT_EQ(ECObjectsStatus::Success, ClassInstance->SetValue("Model", ECN::ECValue(DgnModel::RepositoryModelId())));
 ASSERT_EQ(ECObjectsStatus::Success, ClassInstance->SetValue("Parent", ECN::ECValue(rootSubject->GetElementId(), relClass->GetId())));
}
else if ((className == BIS_CLASS_ExternalSource) || (className == BIS_CLASS_ExternalSourceAttachment) || (className == BIS_CLASS_ExternalSourceGroup))
{
 ASSERT_EQ(ECObjectsStatus::Success, ClassInstance->SetValue("Model", ECN::ECValue(DgnModel::RepositoryModelId())));
}
else
{
 ASSERT_EQ(ECObjectsStatus::Success, ClassInstance->SetValue("Model", ECN::ECValue(linkModel->GetModelId())));
}
```
For above mentioned 4 classes, the default case of “LinkModel” is being used. Now @diegoalexdiaz has introduced a new partition `SheetIndexPartition` and a model `SheetIndexModel` that he wants to use for these classes that's why we wanted to provide support for it. 

Is there any existing option that can be consumed here?

**Comment by diegoalexdiaz** [comment]
@naveedkhan8067, you don't need to introduce C++ handlers to address the new partition+model classes. The test code can use the generic APIs in iModelPlatform to instantiate them.

```
ECN::IECInstance ecInstance = ... 
DgnElementPtr partitionElement = db.Elements().CreateElement(ecInstance);

ECN::IECInstance ecInstance = ...
DgnModelPtr model = db.Models().CreateModel(stat, ecInstance);
```

In both cases, the needed pieces (e.g. properties and/or class) to instantiate those objects are captured by the ecInstance - so please look up in other test-cases how to set up those ecInstances.

**Comment by diegoalexdiaz** [comment]
@naveedkhan8067 , please abandon this PR. 

Its initial motivation ([PR on bis-schemas](https://github.com/iTwin/bis-schemas/pull/196) failing to pass a couple of tests) was already addressed by a different PR in imodel-native ([link](https://github.com/iTwin/imodel-native/pull/269)) that was recently merged.

**Pull request closed without merging**